spec/lib/nhs/immunisations_api_spec.rb

```diff
@@ -643,4 +677,114 @@
       end
     end
   end
+
+  describe "search_immunisations" do
+    subject(:perform_request) do
+      described_class.search_immunisations(
+        patient,
+        programmes:,
+        date_from:,
+        date_to:
+      )
+    end
+
+    let(:nhs_number) { "9449308357" }
+    let(:programmes) { [create(:programme, :flu)] }
+    let(:date_from) { Time.new(2025, 8, 1, 12, 30, 37, "+01:00") }
+    let(:date_to) { Time.new(2025, 10, 1, 10, 35, 32, "+01:00") }
+
+    let!(:request_stub) do
+      stub_request(
+        :get,
+        "https://sandbox.api.service.nhs.uk/immunisation-fhir-api/FHIR/R4/Immunization"
+      ).with(query: expected_query).to_return(status:, body:, headers:)
+    end
+
+    let(:expected_query) do
+      {
+        "patient.identifier" =>
+          "https://fhir.nhs.uk/Id/nhs-number|#{patient.nhs_number}",
+        "-immunization.target" => "FLU",
+        "-date.from" => "2025-08-01",
+        "-date.to" => "2025-10-01"
+      }
+    end
+
+    let(:status) { 200 }
+    let(:body) do
+      File.read(
+        Rails.root.join("spec/fixtures/fhir/search_response_2_results.json")
+      )
+    end
```

Comment:
If you put fixtures in `spec/fixtures/files`, you can [use the `file_fixture` helper method](https://rspec.info/features/6-0/rspec-rails/file-fixture/), this is the more standard approach when using with RSpec fixtures. I know we've got a few fixtures not in `spec/fixtures/files`, but it's one of the many things on my to do list to fix one day. 😅 


```suggestion
    let(:body) { file_fixture("fhir/search_response_2_results.json").read }
```

Reply:
Am I right in thinking that this would be using a new Gem? I can't see `file_fixture` used anywhere else in Mavis currently

Reply:
Here's an example of it being used: https://github.com/nhsuk/manage-vaccinations-in-schools/blob/8b814db86a61f49259c9a2ec99f5ac64f9bef3f0/spec/lib/nhs/pds_spec.rb#L13